inference-engine/src/transformations/src/transformations/convert_precision.cpp

```diff
@@ -0,0 +1,211 @@
+        if (to == element::i32 || to == element::i64) {
+            shapeof->set_output_type(to);
+            return true;
+        }
+    }
+    return false;
+}
+
+bool fuse_type_to_parameter(std::shared_ptr<Node> node, element::Type to, size_t idx) {
+    if (auto param = as_type_ptr<opset3::Parameter>(node)) {
+        param->set_element_type(to);
+        return true;
+    }
+    return false;
+}
+
+bool fuse_type_to_convert(std::shared_ptr<Node> node, element::Type to, size_t idx) {
+    if (auto convert = as_type_ptr<opset3::Convert>(node)) {
+        convert->set_convert_element_type(to);
+        return true;
+    }
+    return false;
+}
+
+bool fuse_type_to_nms3(std::shared_ptr<ngraph::Node> node, ngraph::element::Type to, size_t idx) {
+    if (auto nms = as_type_ptr<opset3::NonMaxSuppression>(node)) {
+        nms->set_output_type(to);
+        return true;
+    }
+    return false;
+}
+
+bool fuse_type_to_nms4(std::shared_ptr<ngraph::Node> node, ngraph::element::Type to, size_t idx) {
+    if (auto nms = as_type_ptr<opset4::NonMaxSuppression>(node)) {
+        nms->set_output_type(to);
+        return true;
+    }
+    return false;
+}
+
+bool fuse_type_to_topk(std::shared_ptr<ngraph::Node> node, ngraph::element::Type to, size_t idx) {
+    if (auto topk = as_type_ptr<opset4::TopK>(node)) {
+        if (idx == 1 && (to == element::i32 || to == element::i64)) {
+            topk->set_index_element_type(to);
+            return true;
+        }
+    }
+    return false;
+}
+
+bool fuse_type_to_nonzero(std::shared_ptr<ngraph::Node> node, ngraph::element::Type to, size_t idx) {
+    if (auto nonzero = as_type_ptr<opset4::NonZero>(node)) {
+        if (to == element::i32 || to == element::i64) {
+            nonzero->set_output_type(to);
+            return true;
+        }
+    }
+    return false;
+}
+
+bool fuse_type_to_bucketize(std::shared_ptr<ngraph::Node> node, ngraph::element::Type to, size_t idx) {
+    if (auto b = as_type_ptr<opset4::Bucketize>(node)) {
+        if (to == element::i32 || to == element::i64) {
+            b->set_output_type(to);
+            return true;
+        }
+    }
+    return false;
+}
+
+template <element::Type_t PREC_FROM, element::Type_t PREC_TO>
+std::shared_ptr<Node> change_constant_precision(std::shared_ptr<opset3::Constant> constant) {
+    using src_type = typename element_type_traits<PREC_FROM>::value_type;
+    using dst_type = typename element_type_traits<PREC_TO>::value_type;
+
+    std::vector<src_type> data(std::move(constant->get_vector<src_type>()));
+    std::vector<dst_type> final_data;
+    std::transform(data.begin(), data.end(), std::back_inserter(final_data),
+                   [](src_type val) {
+                       if (val > std::numeric_limits<dst_type>::max()) {
+                           return std::numeric_limits<dst_type>::max();
+                       } else {
+                           return static_cast<dst_type>(val);
```

Comment:
Is there any chance to use more generic and accurate `saturate_cast` from `PrecisionUtils`?

Reply:
Transformation library has no dependency to ie and links only with ngraph. So I would prefer to keep this library property.

Reply:
Moreover ngraph has it's own conversion for float types that we use and I would move this precision utils to ngraph rather than have it as a separate component.

Reply:
> I would move this precision utils to ngraph

It's a good solution as well